tests/unit/amo/sagas/test_addonsByAuthors.js

```diff
@@ -0,0 +1,112 @@
+import SagaTester from 'redux-saga-tester';
+
+import addonsByAuthorsReducer, {
+  fetchOtherAddonsByAuthors,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import addonsByAuthorsSaga from 'amo/sagas/addonsByAuthors';
+import { ADDON_TYPE_THEME, CLEAR_ERROR } from 'core/constants';
+import * as searchApi from 'core/api/search';
+import apiReducer from 'core/reducers/api';
+import {
+  createAddonsApiResult,
+  dispatchClientMetadata,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockSearchApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockSearchApi = sinon.mock(searchApi);
+    sagaTester = new SagaTester({
+      initialState: dispatchClientMetadata().state,
+      reducers: {
+        api: apiReducer,
+        addonsByAuthors: addonsByAuthorsReducer,
+      },
+    });
+    sagaTester.start(addonsByAuthorsSaga);
+  });
+
+  function _fetchOtherAddonsByAuthors(params) {
+    sagaTester.dispatch(fetchOtherAddonsByAuthors({
+      errorHandlerId: errorHandler.id,
+      addonType: ADDON_TYPE_THEME,
+      ...params,
+    }));
+  }
+
+  it('calls the API to retrieve other add-ons', async () => {
+    const addons = [fakeAddon];
+    const slug = fakeAddon.slug;
+    const authors = ['mozilla', 'johnedoe'];
+
+    mockSearchApi
+      // We use `default` because `search` is the default exported function in
```

Comment:
Oh, that sucks. You *can* import it from `core/api` but otherwise we should fix this... should we export it twice or something? This is pretty inelegant-looking.

Reply:
For some reasons, I could not mock the `core/api` ... Maybe because I also require `core/api/search` in the code.